### PR TITLE
Fix listing update notification

### PR DESCRIPTION
**Pull request by rhnonose**
- Fix wrong pattern-match in `emails/server.ex` and test `handle_info` call rather than `handle_cast` call
- Refactor activation to use `PubSub.pubish_update` with changeset
- Rename price and status history modules

### CODE REVIEW

File: apps/re_integrations/lib/notifications/emails/user.ex [hunk 1]
```diff
@@ -88,7 +88,7 @@ defmodule ReIntegrations.Notifications.Emails.User do
                   <a href=\"#{listing_url}\">Imóvel</a>")
   end
 
-  def listing_updated(%User{name: name, email: email}, %Listing{} = listing, changes) do
+  def listing_updated(%Listing{user: %User{name: name, email: email}} = listing, changes) do
```
Comment (joaodubas) [inline apps/re_integrations/lib/notifications/emails/user.ex]: This is the only method that doesn't match a `User` and a `Listing`. Does it make sense to do this change?

**Comment by rhnonose** [comment]
Usually, plurals like `Users` and `Listings` are for context modules, which is a concept sort of introduced in Phoenix 1.3 and gets some inspiration from Domain-Driven Design.

Previously, Phoenix would generate code inside REST controllers, so there was no "business-level" abstraction in its basic structure.

File: apps/re_integrations/lib/notifications/emails/user.ex (hunk 1 above)
Comment (rhnonose) [reply apps/re_integrations/lib/notifications/emails/user.ex]: It's kind of a defensive coding and a little bit old so it's not strictly necessary. Do you think we should remove it?

Comment (joaodubas) [reply apps/re_integrations/lib/notifications/emails/user.ex]: I don't see a reason to remove it. Maybe we could add another PR, changing [`listing_added_admin`](https://github.com/emcasa/backend/blob/744cba5a46ab5f0c1536f6ff850f44f7958748e0/apps/re_integrations/lib/notifications/emails/user.ex#L74-L89) to keep the same matching.

Comment (rhnonose) [reply apps/re_integrations/lib/notifications/emails/user.ex]: Nice, you can create a task in notion with `refactoring` for that :)

**Pull request merged**